internal/cli/command/cluster/integratedservice/services/restore/create.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restore
+
+import (
+	"context"
+
+	"emperror.dev/errors"
+	"github.com/banzaicloud/banzai-cli/.gen/pipeline"
+	"github.com/banzaicloud/banzai-cli/internal/cli"
+	clustercontext "github.com/banzaicloud/banzai-cli/internal/cli/command/cluster/context"
+	"github.com/banzaicloud/banzai-cli/internal/cli/input"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+type createOptions struct {
+	clustercontext.Context
+
+	backupName string
+}
+
+func newRestoreCreateCommand(banzaiCli cli.Cli) *cobra.Command {
+	options := createOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{"c"},
+		Short:   "Restore backup into a new cluster.",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			cmd.SilenceErrors = true
+
+			if err := options.Init(args...); err != nil {
+				return errors.WrapIf(err, "failed to initialize options")
+			}
+
+			return runCreate(banzaiCli, options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVarP(&options.backupName, "backupName", "", "", "Backup name")
+	options.Context = clustercontext.NewClusterContext(cmd, banzaiCli, "create")
+
+	return cmd
+}
+
+func runCreate(banzaiCli cli.Cli, options createOptions) error {
+	client := banzaiCli.Client()
+	orgID := banzaiCli.Context().OrganizationID()
+	clusterID := options.ClusterID()
+
+	if options.backupName == "" {
+		if banzaiCli.Interactive() {
+			restore, err := askBackup(client, orgID, clusterID)
+			if err != nil {
+				return errors.WrapIf(err, "failed to ask restore")
+			}
+
+			options.backupName = restore.Name
+		} else {
+			return errors.NewWithDetails("invalid backup name", "backupName", options.backupName)
+		}
+	}
+
+	_, _, err := client.ArkRestoresApi.CreateARKRestore(context.Background(), orgID, clusterID, pipeline.CreateRestoreRequest{
+		BackupName: options.backupName,
+	})
+	if err != nil {
+		return errors.WrapIfWithDetails(err, "failed to create restore", "clusterID", clusterID, "backupName", options.backupName)
+	}
+
+	log.Info("Restore started to create")
```

Comment:
**Optional**: opinionated
```suggestion
	log.Info("Restore creation started")
```

Reply:
```suggestion
	log.Infof("Starting to restore cluster. You can check the status with `banzai cluster restore result %q`", fixme)
```

Reply:
Yeah, I misinterpreted the context, thanks for the improvement, it's much better this way.